### DISCUSSION TITLE
support multiple remote-bag-urls

### DISCUSSION THREAD

**Discussion by cheery550** [foxglove/community#314] [Q&A]
Since there are many sensors in automatic perception and the bag will become bigger and bigger,  we break data in different topics, like lidar point cloud, multiple videos streams, many objects from radar/camera/fusion. 

Whether it is possible to support multiple remote bags? 
At the same time,  we notice that webviz has support two remote bags. Is it possible to learn from their practices.

**Comment by jtbandes** [comment]
Related: https://github.com/foxglove/community/discussions/123 (sequential bag files), https://github.com/foxglove/community/discussions/160 (comparing bag files)

**Comment by TommeTao** [comment]
This is also not possible with local files, like MCAP files, right?

I have a use case where I want to load multiple MCAP files containing different topics (see also https://github.com/foxglove/community/issues/109#issuecomment-1354209528). I am used to work with CAN bus tools that come with this feature. Therefore, I assumed that I could also load multiple log files into Foxglove Studio and designed my MCAP converter accordingly, i.e. create one file per topic, only to find that I am not able to load those file into the same session.

**Reply by defunctzombie** [reply]
This is not possible with local files. There are two approaches we can suggest as workaround until there is multiple file support:

1. Use `mcap merge` to merge the files into 1 file and open that.
2. For our data platform users, you can upload all your files (with overlapping timerange) and then stream from data platform and that will handle merging the data

**Comment by oosthoek** [comment]
Hello. I am wondering if this is currently being worked on and if so any idea a timeframe?

**Reply by banisadr** [reply]
This is not currently being worked on. We are interested in adding support for comparison between two recordings #160 in the future, but no timeline I can share. Your best option for now would continue to be using `mcap merge` or leveraging our Data Platform.
